File: app/src/main/java/com/haker/ah/SunnyAdapter.java
package com.haker.ah;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

/**
 * Created by js on 2018-05-28.
 */

public class SunnyAdapter extends RecyclerView.Adapter<SunnyAdapter.MyViewHolder>{
    private ArrayList<SunnyData> mDataset;

    public class MyViewHolder extends RecyclerView.ViewHolder {
        public TextView name, lan, lon;

        //ViewHolder
        public MyViewHolder(View view) {
            super(view);
            name = (TextView) view.findViewById(R.id.name);
        }
    }

    public SunnyAdapter(ArrayList<SunnyData> myData){
        this.mDataset = myData;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.sunny_item, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull SunnyAdapter.MyViewHolder holder, int position) {

        holder.name.setText(mDataset.get(position).getName());


    }

    @Override
    public int getItemCount() {
        return mDataset.size();
    }
}

File: app/src/main/java/com/example/alt_map/SunnyAdapter.kt
package com.example.alt_map

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class SunnyAdapter(val context: Context) : RecyclerView.Adapter<SunnyAdapter.Holder>() {

    var data = mutableListOf<SunnyData>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        val view = LayoutInflater.from(context).inflate(R.layout.sunny_item, parent, false)
        return Holder(view)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        val sunnyData = data[position]
        holder.bind(sunnyData)
    }

    override fun getItemCount(): Int {
        return data.size
    }

    inner class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val name = itemView.findViewById<TextView>(R.id.sunny_activity_text)
        val img = itemView.findViewById<ImageView>(R.id.profile_img)
        fun bind(sunnyData: SunnyData) {
            name.text = sunnyData.name
            img.setImageResource(sunnyData.profile)
            img.clipToOutline = true
        }
    }
}